### DISCUSSION TITLE
Geochemistry module question

### DISCUSSION THREAD

**Discussion by Fruitwind** [idaholab/moose#25096] [Q&A Modules: General]
Dear MOOSE team,

we are  learning to apply the geochemistry module with the self-developed application to run 1D calcite dissolution with acidic injected water (pH=4).  I have two questions about applying the module and am grateful for any input or suggestion:
1) The example is on the developing stage, but I faced the issue with being able to run the simulations with the version  of geochemistry module of 30.04 and  not being able to run it with the version of 30.07. 

2) While running the simulations, I receive negative concentrations on the first time steps which obviously should not happen.

Would you have any idea what could be the reason for both of these issues?

I am providing below the geochemistry input file:
[](url)
[chem_THC1D.txt](https://github.com/idaholab/moose/files/12251742/chem_THC1D.txt)

Thanks a lot and a great day,
Elena

**Comment by GiudGiud** [comment]
@WilkAndy @rpodgorney

**Reply by Fruitwind** [reply]
Dear @WilkAndy @rpodgorney, dear @GiudGiud , 

I have been trying to resolve the abovementioned issue with negative concentrations of H+ with unfortunately no success, and I would highly appreciate any advice on how to proceed with it. In addition, I noticed reported negative concentration in one of examples given in MOOSE-Geochemistry simulations: table 9 in:
https://mooseframework.inl.gov/modules/geochemistry/tests_and_examples/geotes_weber_tensleep.html#table:model_bulk
Concentrations of H+ and O2(aq) are negative. Could you please provide some comment on that?

Many thanks and have a great day,
Elena

**Reply by GiudGiud** [reply]
Hello

Did you try the Bounds system? That forces variables to stay within a range
https://mooseframework.inl.gov/source/auxkernels/ConstantBoundsAux.html

**Comment by Fruitwind** [comment]
upd. Not sure if I need to make another post, but when I do the Multiapp copy tranfer with the transport, the geochemistry module simulation overwrites the boundary condition in the transport (Dirichlet BC with the constant concentrations)

**Reply by GiudGiud** [reply]
that s normal. Transfers can overwrite inital conditions for a variable. 
If you dont want that, transfer with an execute_on of TIMESTEP_END

**Reply by Fruitwind** [reply]
Thank you! The fact is, we are using the Dirichlet boundary condition with the fixed concentration, and the fixed concentration is overwritten.

**Reply by Fruitwind** [reply]
preset option is on and we do the TIMESTEP_END transfer

**Reply by GiudGiud** [reply]
can you apply the DirichletBC in the app that provides the data for the transfer too?

**Reply by Fruitwind** [reply]
we use Geochemistry module of MOOSE to update the reactions and send it to the transport simulation. The BC is set in the transport app, while it cannot be set up in the Geochemistry module. The problem is, the BC for the solute transport in the transport app are overwritten by Geochemistry.

**Reply by Fruitwind** [reply]
update: we set up transfer to  Geochemistry TIMESTEP_BEGINNING and it seems to work. The issues above keep being a problem.

**Reply by MengnanLi91** [reply]
Have you tried to run a batch simulation with just Geochemistry module?  It may be easier to figure out the problem. One issue I find with your constraint is that you don't actually  have a fixed pH=4 in your setup. If you have a fixed pH for your chemical reaction, you should reflect it using `constraint_species` and `constraint_value`, usually is the `log10activity ` of `H+`
